command/agent/job_endpoint.go

```diff
@@ -710,16 +712,32 @@ func ApiTgToStructsTG(taskGroup *api.TaskGroup, tg *structs.TaskGroup) {
 		}
 	}
 
-	if taskGroup.Update != nil {
+	// Convert the UpdateStrategy, using defaults from the job
+	if taskGroup.Update != nil || jobUpdate != nil {
+		var au *api.UpdateStrategy
+		if jobUpdate != nil {
+			au = jobUpdate.Copy()
+			au.Merge(taskGroup.Update)
+		} else {
+			au = taskGroup.Update
+		}
```

Comment:
I suspect the merging of jobUpdate might be a subtle change in behavior.  How about use taskGroup.Update if it's not nil, otherwise use `jobUpdate`?  Either way, we should note it in the docs/changelog/release notes.